tests/pipelines/test_pipelines_automatic_speech_recognition.py

```diff
@@ -674,6 +674,56 @@ def test_whisper_timestamp_prediction(self):
             },
         )
 
+    @slow
+    @require_torch
+    def test_whisper_word_timestamps_batched(self):
+        pipe = pipeline(
+            task="automatic-speech-recognition",
+            model="openai/whisper-tiny",
+            chunk_length_s=3,
+            return_timestamps="word",
+        )
+        data = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
+        sample = data[0]["audio"]
+
+        # not the same output as test_simple_whisper_asr because of chunking
+        EXPECTED_OUTPUT = {
+            "text": " Mr. Quilder is the apostle of the middle classes and we are glad to welcome his gospel.",
+            "chunks": [
+                {"text": " Mr.", "timestamp": (0.48, 0.96)},
+                {"text": " Quilder", "timestamp": (0.96, 1.24)},
+                {"text": " is", "timestamp": (1.24, 1.5)},
+                {"text": " the", "timestamp": (1.5, 1.72)},
+                {"text": " apostle", "timestamp": (1.72, 1.98)},
+                {"text": " of", "timestamp": (1.98, 2.32)},
+                {"text": " the", "timestamp": (2.32, 2.5)},
+                {"text": " middle", "timestamp": (2.5, 2.68)},
+                {"text": " classes", "timestamp": (2.68, 3.2)},
+                {"text": " and", "timestamp": (3.2, 3.56)},
+                {"text": " we", "timestamp": (3.56, 3.68)},
+                {"text": " are", "timestamp": (3.68, 3.8)},
+                {"text": " glad", "timestamp": (3.8, 4.1)},
+                {"text": " to", "timestamp": (4.1, 4.34)},
+                {"text": " welcome", "timestamp": (4.3, 4.6)},
+                {"text": " his", "timestamp": (4.6, 4.94)},
+                {"text": " gospel.", "timestamp": (4.94, 5.82)},
+            ],
+        }
+
+        # batch size 1: copy the audio sample since pipeline consumes it
+        output = pipe(sample.copy(), batch_size=1)
+        self.assertDictEqual(
+            output,
+            EXPECTED_OUTPUT,
+        )
+
+        # batch size 2
+        output = pipe(sample, batch_size=2)
```

Comment:
Just to make sure I've understood here. In this case, `sample` will effectively be loaded twice in the dataloader to create a batch_size of 2. 

Is there a reason for not using two different samples i.e. `sample = data[:2]["audio"]` ?

Reply:
If I understood correctly, if `chunk_length_s<len(audio)`, then the audio is split into a certain number of chunks, which then computed in a batch if  `batch_size>1`.

So in any case, this test if the batch feature works correctly!
 

cc @sanchit-gandhi, is that correct ?

Reply:
Ah, OK, makes sense. Could you add a comment explaining that here? Otherwise it's not obvious what the test is doing